File: leetcode/leetcode/Leetcode-121.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        length = len(prices)
        if length == 0:
            return 0
        m_min = prices[0]
        m_max = 0
        for i in prices:
            m_min = min(m_min, i)
            m_max = max(m_max, i - m_min)
        return m_max

File: leetcode/leetcode/Leetcode-121.cpp
#include<vector.h>
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() == 0)
        {
            return 0;
        }
        int min = prices[0];
        int max = 0;
        for (int i = 1; i < prices.size(); i++)
        {
            if (min < prices[i])
            {
                max = max > (prices[i] - min) ? max : (prices[i] - min);
            }
            else
            {
                min = prices[i];
            }
            
        }
        return max;
    }
};